From d07061c8f6c3f78eed8dd1bc2f755609cd80d96f Mon Sep 17 00:00:00 2001
From: Xianzhong <b07117@freescale.com>
Date: Tue, 19 Jan 2016 16:18:21 +0800
Subject: [PATCH] MGS-1560 [#2172] fix gpu kernel crash with invalid pointer

it is necessary to check the node object pointer before use,
otherwise the board will reboot with kernel crash.

Date: Jan 19, 2016
Signed-off-by: Xianzhong <xianzhong.li@nxp.com>
---
 drivers/mxc/gpu-viv/hal/kernel/gc_hal_kernel.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/mxc/gpu-viv/hal/kernel/gc_hal_kernel.c b/drivers/mxc/gpu-viv/hal/kernel/gc_hal_kernel.c
index 0d9a990..2c8b2f9 100644
--- a/drivers/mxc/gpu-viv/hal/kernel/gc_hal_kernel.c
+++ b/drivers/mxc/gpu-viv/hal/kernel/gc_hal_kernel.c
@@ -2426,6 +2426,16 @@ gckKERNEL_Dispatch(
                 Interface->u.Cache.node,
                 &nodeObject));
 
+            /* Verify the nodeObject. */
+            if ((nodeObject == gcvNULL)
+            ||  (nodeObject->node == gcvNULL)
+            ||  (nodeObject->node->VidMem.memory == gcvNULL)
+            )
+            {
+                /* Invalid object. */
+                gcmkONERROR(gcvSTATUS_INVALID_OBJECT);
+            }
+
             if (nodeObject->node->VidMem.memory->object.type == gcvOBJ_VIDMEM
              || nodeObject->node->Virtual.contiguous
             )
-- 
1.8.0

